yt_dlp/extractor/rtrfm.py

```diff
@@ -0,0 +1,69 @@
+from __future__ import unicode_literals
+
+from .common import InfoExtractor
+
+
+class RTRFMIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?rtrfm.com.au/(?:shows/(?P<show1>[^/]+)|show-episode/(?P<show2>[^-]+)-(?P<date>[0-9]{4}-[0-9]{2}-[0-9]{2})/?)'
+    _PLAY_SHOW = r"\.playShow\('(?P<show>[^']+)', *'(?P<date>[0-9]{4}-[0-9]{2}-[0-9]{2})', *'(?P<title>[^']+)'\)"
+    _PLAY_SHOW_FROM = r"\.playShowFrom\('(?P<show>[^']+)', *'(?P<date>[0-9]{4}-[0-9]{2}-[0-9]{2})', *'?P<title>([^']+)', \d+\)"
+    _RESTREAMS_URL = 'https://restreams.rtrfm.com.au/rzz'
+    _TESTS = [
+        {
+            'url': 'https://rtrfm.com.au/shows/breakfast/',
+            # the downloaded file changes daily so the md5 is not checkable
+            'info_dict': {
+                'id': r're:^breakfast-\d{4}-\d{2}-\d{2}$',
+                'ext': 'mp3',
+                'series': 'Breakfast with Taylah',
+                'title': r're:^Breakfast with Taylah \d{4}-\d{2}-\d{2}$',
+                'description': 'md5:0979c3ab1febfbec3f1ccb743633c611',
+            },
+            'skip': 'tests with re in the id expect the re in the filename '
+                    'instead of the id, but the id depends on the date.',
+        },
+        {
+            'url': 'https://rtrfm.com.au/show-episode/breakfast-2021-11-11/',
+            'md5': '396bedf1e40f96c62b30d4999202a790',
+            'info_dict': {
+                'id': 'breakfast-2021-11-11',
+                'ext': 'mp3',
+                'series': 'Breakfast with Taylah',
+                'title': 'Breakfast with Taylah 2021-11-11',
+                'description': 'md5:0979c3ab1febfbec3f1ccb743633c611',
+            },
+        },
+        {
+            'url': 'https://rtrfm.com.au/show-episode/breakfast-2020-06-01/',
+            'md5': '594027f513ec36a24b15d65007a24dff',
+            'info_dict': {
+                'id': 'breakfast-2020-06-01',
+                'ext': 'mp3',
+                'series': 'Breakfast with Taylah',
+                'title': 'Breakfast with Taylah 2020-06-01',
+                'description': r're:^Breakfast with Taylah ',
+            },
+            'params': {
+                # This audio has expired
+                'skip_download': True,
+            },
+        },
+    ]
+
+    def _real_extract(self, url):
+        show1, show2, date = self._match_valid_url(url).groups()
+        webpage = self._download_webpage(url, show1 or show2)
```

Comment:
Since you only appear to be using show1/show2 for the `video_id` param of download_webpage (and we overwrite the date anyways), we can probably simplify the _VALID_URL regex.
```suggestion
        display_id = self._match_id(url)
        webpage = self._download_webpage(url, display_id)
```

Reply:
Adopted.